homeassistant/components/prometheus/__init__.py

```diff
@@ -318,6 +323,32 @@ def _handle_climate(self, state):
                     float(action == current_action)
                 )
 
+    def _handle_humidifier(self, state):
+        humidity = state.attributes.get(ATTR_HUMIDITY)
+        if humidity:
+            metric = self._metric(
+                "humidity", self.prometheus_cli.Gauge, "Target Relative Humidity"
+            )
+            metric.labels(**self._labels(state)).set(humidity)
+
+        metric = self._metric(
+            "humidifier_state",
+            self.prometheus_cli.Gauge,
+            "State of the humidifier (0/1)",
+        )
+        try:
+            value = self.state_as_number(state)
+        except ValueError:
+            value = 0.0
+        metric.labels(**self._labels(state)).set(value)
+
+        mode = state.attributes.get(ATTR_MODE)
+        modes = state.attributes.get(ATTR_AVAILABLE_MODES)
+        if mode and modes:
+            metric = self._metric("mode", self.prometheus_cli.Gauge, "Humidifier Mode")
+            value = modes.index(mode)
```

Comment:
How would the user be able to determine which index means what mode?

Please export `mode` as a metric label, and use the value of `1` for the current mode, exporting `0` for other modes. An example of how to do this is the `climate_action` metric.